libvirt/utils_libvirt.go

```diff
@@ -68,3 +70,70 @@ func getHostArchitecture(virConn *libvirt.Connect) (string, error) {
 
 	return capabilities.Host.CPU.Arch, nil
 }
+
+type GuestCapabilities struct {
+	XMLName xml.Name `xml:"capabilities"`
+	Architectures []struct {
+		Name string `xml:"name,attr"`
+		Wordsize string `xml:"wordsize"`
+		Emulator string `xml:"emulator"`
+		Machines []Machine `xml:"machine"`
+	} `xml:"guest>arch"`
+
+}
+
+type Machine struct {
+	Name string `xml:",chardata"`
+	Canonical string `xml:"canonical,attr"`
+}
+
+
+
+
```

Comment:
A single empty line is enough.

Reply:
Thanks, I'll review and update.

Reply:
@monstermunchkin  Hopefully a cleaner version now